libs/helm/autocomplete/src/lib/hlm-autocomplete.ts

```diff
@@ -0,0 +1,244 @@
+				<hlm-autocomplete-search hlmAutocompleteTrigger [disabled]="!_search()">
+					<ng-icon name="lucideSearch" hlm />
+					<input
+						#input
+						type="text"
+						autocomplete="off"
+						hlm-autocomplete-search-input
+						[id]="inputId()"
+						[placeholder]="placeholderText()"
+						[disabled]="_disabled()"
+						[value]="_search()"
+						(input)="_onSearchChanged($event)"
+						(keydown.arrowDown)="_setPopoverState('open')"
+					/>
+
+					<button
+						class="flex items-center justify-center outline-none disabled:cursor-not-allowed [&>_ng-icon]:opacity-50"
+						tabindex="-1"
+						type="button"
+						aria-label="Toggle dropdown"
+						[disabled]="_disabled()"
+						(click)="_toggleOptions()"
+					>
+						<ng-icon name="lucideChevronDown" hlm />
+					</button>
+				</hlm-autocomplete-search>
+
+				<div
+					*brnPopoverContent="let ctx"
+					hlmPopoverContent
+					class="max-h-60 overflow-y-auto p-0"
+					[style.width.px]="_elementRef.nativeElement.offsetWidth"
+				>
+					<hlm-command-list>
+						<hlm-command-group>
+							@for (option of _filteredOptions(); track option) {
+								<button hlm-command-item [value]="option" (selected)="_optionSelected(option)">
+									{{ option }}
+								</button>
+							}
+						</hlm-command-group>
+					</hlm-command-list>
+
+					<div *brnCommandEmpty hlmCommandEmpty>
+						@if (loading()) {
+							<ng-content select="[loading]">Loading...</ng-content>
+						} @else {
+							<ng-content select="[empty]">No options found</ng-content>
+						}
+					</div>
+				</div>
+			</hlm-autocomplete-command>
+		</brn-popover>
+	`,
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	host: {
+		'[class]': '_computedClass()',
+	},
+})
+export class HlmAutocomplete implements ControlValueAccessor {
+	private readonly _brnPopover = viewChild.required(BrnPopover);
+	private readonly _inputRef = viewChild.required('input', { read: ElementRef });
+
+	protected readonly _elementRef = inject(ElementRef<HTMLElement>);
+
+	public readonly userClass = input<ClassValue>('', { alias: 'class' });
+	protected readonly _computedClass = computed(() => hlm('block w-full', this.userClass()));
+
+	/** The list of options to display in the autocomplete. */
+	public readonly options = input<string[]>([]);
+
+	/** The selected value. */
+	public readonly value = input<string>();
+	protected readonly _value = linkedSignal(() => this.value());
+
+	/** The default filter function */
+	private readonly _defaultFilter = (value: string, search: string) =>
+		value.toLowerCase().includes(search.toLowerCase());
+
+	/** A custom filter function to determine which items are visible when searching. */
+	public readonly filter = input<CommandFilter>(this._defaultFilter);
+
+	/** The search query. */
+	public readonly search = input<string>('');
+	protected readonly _search = linkedSignal(() => this.search());
+
+	protected readonly _filteredOptions = computed(() =>
```

Comment:
@ashley-hunter I am thinking about changing `_filteredOptions` to an input to replace `options`. This would move the filter obligation to the ui when the autocomplete is used. What do you think about that? Do you think this makes it more complicated? Otherwise we might need another input for this filter.